### PR TITLE
Reconnection tests [12522]

### PR DESCRIPTION
**Pull request by rsanchez15**
Reverts eProsima/Discovery-Server#44.

This PR restore the changes made in https://github.com/eProsima/Discovery-Server/pull/41. 
Do not merge until https://github.com/eProsima/Fast-DDS/pull/2247 is merged.

### CODE REVIEW

**Comment by rsanchez15** [comment]
This PR is approved in favor of #41.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]
Partial Review pending xml and snapshot check

File: src/main.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #include "log/DSLog.h"
 #include "version/config.h"
 
-#if FASTRTPS_VERSION_MAJOR >= 2 && FASTRTPS_VERSION_MINOR >= 1
+#if FASTRTPS_VERSION_MAJOR >= 2 || (FASTRTPS_VERSION_MAJOR == 2 && FASTRTPS_VERSION_MINOR >= 1)
 #include <fastdds/dds/log/StdoutErrConsumer.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/main.cpp:19]: :+1:

File: src/main.cpp [hunk 2]
```diff
@@ -40,18 +40,15 @@ int main(
...
-    // Log::SetCategoryFilter(std::regex("(DISCOVERY_SERVER)"));
+    #if LOG_LEVEL_INFO
+        Log::SetVerbosity(Log::Kind::Info);
+    #elif LOG_LEVEL_WARN
+        Log::SetVerbosity(Log::Kind::Warning);
+    #elif LOG_LEVEL_ERROR
+        Log::SetVerbosity(Log::Kind::Error);
+    #else
+        Log::SetVerbosity(Log::Kind::Error);
+    #endif
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/main.cpp:51]: :+1:

**Review by jsantiago-eProsima — APPROVED** [review]
LGTM

**Review by EduPonz — APPROVED** [review]
LGTM

**Review by EduPonz — APPROVED** [review]
LGTM

**Pull request merged**